app/src/main/java/ch/epfl/polychef/utils/OpinionsMiniatureAdapter.java

```diff
@@ -0,0 +1,158 @@
+    public void onBindViewHolder(@NonNull MiniatureViewHolder holder, int position) {
+        Opinion opinion = displayedOpinions.get(position);
+        User user = userOp.get(opinion);
+        holder.rate.setRating(opinion.getRate());
+        holder.commentText.setText(opinion.getComment());
+        holder.commentUsername.setText(userOp.get(opinion).getUsername());
+        int imageID = User.getResourceImageFromUser(userOp.get(opinion));
+        holder.profilePict.setImageResource(imageID);
+    }
+
+    @Override
+    public int getItemCount() {
+        return displayedOpinions.size();
+    }
+
+    class MiniatureViewHolder extends RecyclerView.ViewHolder {
+
+        RatingBar rate;
+        ImageView profilePict;
+        TextView commentText;
+        TextView commentUsername;
+
+        public MiniatureViewHolder(@NonNull View itemView) {
+            super(itemView);
+            rate = itemView.findViewById(R.id.ratingCommentBar);
+            profilePict = itemView.findViewById(R.id.commentProfilePict);
+            commentText = itemView.findViewById(R.id.commentText);
+            commentUsername = itemView.findViewById(R.id.commentUsername);
+        }
+    }
+
+    /**
+     * This class is the listener that when we click on a miniature it swaps to the new fragment with the full recipe displayed.
+     */
+    class MiniatureOnClickListener implements View.OnClickListener {
+
+        RecyclerView recyclerView;
+
+        public MiniatureOnClickListener(RecyclerView recyclerView) {
+            this.recyclerView = recyclerView;
+        }
+
+        @Override
+        public void onClick(View view) {
+            // Get the clicked user from the recyclerView
+            int opinionPosition = recyclerView.getChildLayoutPosition(view);
+            Opinion clickedOpinion = allOpinions.get(opinionPosition);
+            User clickedUser = userOp.get(clickedOpinion);
+            Bundle bundle = new Bundle();
+            bundle.putSerializable("User", clickedUser);
+
+            ((HomePage) mainContext)
+                    .getNavController()
+                    .navigate(R.id.userProfileFragment, bundle);
+        }
+    }
+
+    public void loadNewComments(){
+        isLoading = true;
+        for(int i = currentIndex; i < Math.min(currentIndex + nbOfOpinionsLoadedAtATime, allOpinions.size()); i++){
+            final Opinion current = allOpinions.get(i);
+            final int forIndex = i;
+            userStorage.getUserByID(recipe.getRating().getUserIdFromOpinion(current), new CallHandler<User>() {
+                @Override
+                public void onSuccess(User user) {
+                    userOp.put(current, user);
+                    displayedOpinions.add(current);
+                    notifyDataSetChanged();
+                    if(forIndex == Math.min(currentIndex + nbOfOpinionsLoadedAtATime, allOpinions.size()) - 1){
+                        isLoading = false;
+                    }
+                }
```

Comment:
Could use the MultipleCallHandler, just like in UserListFragment.
I can see an edge case where the query for the last comment comes back early and allow loading too soon (I don't think it's a huge problem).

Since there is a similar thing in UserProfileFragment, we could change it next sprint if we have time.